### PR TITLE
Add support for testing reasons.

### PR DESCRIPTION
**Pull request by dmach**

### CODE REVIEW

File: dnf-behave-tests/features/steps/history.py [hunk 1]
```diff
@@ -120,3 +122,26 @@ def step_impl(context):
                     '[history] package "{0}" matched as userinstalled.'.format(package))
         else:
             raise ValueError('Invalid action "{0}".'.format(action))
+
+
+@then('reasons are following')
+def step_impl(context):
+    check_context_table(context, ["Reason", "Package"])
+    cmd = " ".join(context.dnf.get_cmd(context) + ["repoquery", "--installed", "--qf='NEVRA:%{name}-%{epoch}:%{version}-%{release}.%{arch} REASON:%{reason}'"])
+    _, cmd_stdout, _ = run(cmd, shell=True, can_fail=True)
```
Comment (lukash) [inline dnf-behave-tests/features/steps/history.py]: I think you should either use `can_fail=False` or handle failure yourself here.

File: dnf-behave-tests/features/steps/history.py [hunk 2]
```diff
@@ -120,3 +122,26 @@ def step_impl(context):
...
+
+    re_reason = re.compile(r"^NEVRA:(.*) REASON:(.*)$")
```
Comment (lukash) [inline dnf-behave-tests/features/steps/history.py]: I don't think the `repoquery --qf` command prints anything else than the formatted list. Unless it does, you could use a format like: `'%{name}-%{epoch}:%{version}-%{release}.%{arch} %{reason}'` and then leave out the whole regex and simply split by space to get the dict pair. Constructing the dict could then basically be a oneliner.

File: dnf-behave-tests/features/steps/common/rpmdb.py [hunk 3]
```diff
@@ -10,19 +10,28 @@
...
         if name.startswith("gpg-pubkey"):
             continue
-        decoded = dict(name=name,
-                evr=normalize_epoch(hdr["evr"].decode()),
-                arch=hdr["arch"].decode())
+        decoded = {
+            "name": name,
+            "evr": normalize_epoch(_get_hdr_str(hdr, "evr")),
+            "arch": _get_hdr_str(hdr, "arch"),
+        }
```
Comment (lukash) [inline dnf-behave-tests/features/steps/common/rpmdb.py]: Some description of what's going on would be nice for this commit.

**Comment by lukash** [comment]
@dmach sorry about the conflict...

However, it could have been avoided it if this was resolved in a timely fashion :slightly_smiling_face: Could you rebase and could we get this in?

**Comment by j-mracek** [comment]
My apology, but I am closing it due to inactivity. Please feel free to reopen it after rebase.

**Pull request closed without merging**